ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/websocket/ClientSessions.kt

```diff
@@ -29,3 +33,55 @@ internal class DelegatingClientWebSocketSession(
     override val call: HttpClientCall,
     session: WebSocketSession
 ) : ClientWebSocketSession, WebSocketSession by session
+
+/**
+ * Serializes [data] to frame and enqueue this frame, may suspend if outgoing queue is full. May throw an exception if outgoing channel is already
+ * closed, so it is impossible to transfer any message. Frames that were sent after close frame could be silently
+ * ignored. Please note that close frame could be sent automatically in reply to a peer close frame unless it is
+ * raw websocket session.
+ */
+public suspend inline fun <reified T : Any> DefaultClientWebSocketSession.sendSerialized(data: T) {
+    val serializedData = call.client?.plugin(WebSockets)?.contentConverter?.serialize(
+        charset = call.request.headers.suitableCharset(),
+        typeInfo = typeInfo<T>(),
+        value = data
+    ) ?: throw WebsocketConverterNotFoundException("No converter was found for websocket")
+
+    outgoing.send(serializedData)
+}
+
+/**
+ * Dequeue frame and deserializes to type [T] using websocket content converter.
+ * May throw an exception [WebsocketDeserializeException] if converter can't deserialize frame data to type [T].
+ * May throw [WebsocketDeserializeException] if received frame type is not [Frame.Text] or [Frame.Binary].
+ * In this case [WebsocketDeserializeException.frame] contains received frame.
+ *
+ * May throw [ClosedReceiveChannelException] if channel was closed
+ *
+ * @throws WebsocketConverterNotFoundException if no [contentConverter] is found for the [WebSockets] plugin
+ * @throws WebsocketDeserializeException if received frame can't be deserialized to type [T]
+ */
+public suspend inline fun <reified T : Any> DefaultClientWebSocketSession.receiveDeserialized(): T {
+    val frame = incoming.receive()
+    val data = when (frame) {
```

Comment:
what do you think about adding `fun isApplicable(frame: Frame): Boolean` function to a `ContentConverter` and calling it here?

Reply:
Sounds good, I've implemented it